staging/src/k8s.io/apiserver/pkg/storage/etcd3/metrics/metrics.go

```diff
@@ -60,6 +61,14 @@ var (
 		},
 		[]string{"resource"},
 	)
+	objectDecodeCounter = compbasemetrics.NewCounterVec(
+		&compbasemetrics.CounterOpts{
+			Name:           "etcd_object_decode",
+			Help:           "Counter of stored objects which is decoded for each group, version and kind.",
```

Comment:
What is the purpose of this metric? Given that `decode` method is called from both writes and reads, I can only imagine it being used to estimate the cost related to decoding. Was that your idea? How you would like to use this metric?

Reply:
By exposing this metric, we can tell into which gvk stored object decoded and which deprecated gvk is still in use , these object in deprecated gvk coding should be migrated to the latest gvk.
So this counter metric makes sense with the consideration of these. 
By quering below promeql:
`sum((sum by(group,version)(apiserver_requested_deprecated_apis{} )* on(group,version)
group_right() increase(etcd_object_decode{}[1m]))) by(group,version)`

we will know whether stored object is in depcreated gvk and should be migrated or not.